src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -51,53 +55,119 @@ public boolean isMaintainer() {
     return this.isMaintainer;
   }
 
+  public String getUserEmail() {
+    return this.email;
+  }
+
+  public boolean isLoggedIn() {
+    return this.isLoggedIn;
+  }
+
   //TODO(): get correct moderator status for organization
   public boolean isModerator(long organizationId) {
       return false;
   }
 
-  public static GivrUser getUserByIdFromDatastore(String userId) {
+  /*
+   * This function returns the PreparedQuery for the "User" entities that have the propertyName, propertyValue.
+   * Use case: PreparedQuery preparedQuery = GivrUser.getPreparedQueryResultOfUserWithProperty("userId", "abcdefg");
+   */
+  public static PreparedQuery getPreparedQueryResultOfUserWithProperty(String propertyName, String propertyValue) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Filter queryFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+    Filter queryFilter = null;
+    if (propertyName.equals("isMaintainer")) {
+      queryFilter = new FilterPredicate(propertyName, FilterOperator.EQUAL, Boolean.parseBoolean(propertyValue));
+    } else {
+      queryFilter = new FilterPredicate(propertyName, FilterOperator.EQUAL, propertyValue);
+    }
     Query query = new Query("User").setFilter(queryFilter);
-      
-    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+
     PreparedQuery preparedQuery = datastore.prepare(query);
-    QueryResultList<Entity> userResult = preparedQuery.asQueryResultList(fetchOptions);
+    return preparedQuery;
+  }
 
-    boolean isMaintainer = false;
-    boolean isLoggedIn = true;
+  // Check if User with propertyName, propertyValue exists within Datastore.
+  public static boolean checkIfUserWithPropertyExists(String propertyName, String propertyValue) {
```

Comment:
Entity getUserFromDatabaseWithProperty(String propertyName, String propertyValue) {
  // either return null or throw an exception if it doesn't exist.
}

Reply:
I chose to return null here!